x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -75,7 +77,14 @@ export class TaskRunner {
     const requestHeaders: Record<string, string> = {};
 
     if (apiKey) {
-      requestHeaders.authorization = `ApiKey ${apiKey}`;
+      const subApiKeyResult = await this.getSubApiKey(spaceId, apiKey);
+      if (subApiKeyResult) {
+        const apiKeyHeaderValue = Buffer.from(
+          `${subApiKeyResult.id}:${subApiKeyResult.api_key}`
+        ).toString('base64');
+        requestHeaders.authorization = `ApiKey ${apiKeyHeaderValue}`;
+        this.subApiKeyId = subApiKeyResult.id;
+      }
```

Comment:
In regards to this,  I'm thinking we move the logic within `getApiKeyForAlertPermissions`. The function can return a sub API key within there (also avoids returning an API key that shouldn't be used in this file).

We can also add some retry logic somewhere to handle the scenario it fails to create a sub API key (maybe max attempts 3x?). On failures, all the code in the function would have to re-run (load API key from alert then create sub-api key).